### ISSUE TITLE
Support for cephfs?

### ISSUE THREAD

**Issue by bluepuma77** [olljanat/csi-plugins-for-docker-swarm#4]
Would it be possible to support [cephfs](https://docs.ceph.com/en/quincy/cephfs/index.html) for self-hosting?

Docker hosting [kraudcloud is supporting cephfs](https://docs.kraudcloud.com/blog/2022/01/31/global-file-system-is-now-generally-available/) now, but sadly it's just the Docker CLI, they use their own orchestrator implementation in the backend.

**Comment by s4ke** [comment]
I guess the best course of action is to ask over at https://github.com/ceph/ceph-csi. But yeah, I would appreciate this as well.

**Comment by s4ke** [comment]
Essentially, as long as the CSI provider does not hook strongly into Kubernetes, it is a candidate to be made compatible. Also, having support for Nomad is a good indicator that Swarm will just work (after some repackaging as a plugin).

Sadly, I don't currently have a CEPH cluster to try this on, so I can't help you much. The resources in here are a good starting point though. Also, you can join some of us Swarm Fans over at https://devops.fan , maybe we can help each other out there.

@olljanat maybe its worth to build some "dev support channel" initiative where all the Swarm CSI enthusiasts can gather. I think the linked discord can be that for live chat. Wdyt?

**Comment by bluepuma77** [comment]
> Swarm Fans over at https://devops.fan/

Thanks, didn't know about this. 

I created [reddit.com/r/DockerSwarm](https://www.reddit.com/r/DockerSwarm/), but only 5 members, so interest in Docker Swarm seems very limited.

Sidenote: I just checked official Docker docs "[Extending Docker with plugins](https://docs.docker.com/engine/extend/legacy_plugins/)", seems kind of telling that the URL is "legacy_plugins" and they have Volume plugins listed like [flocker](https://github.com/ScatterHQ/flocker), which is retired since 2018.

**Comment by olljanat** [comment]
> @olljanat maybe its worth to build some "dev support channel" initiative where all the Swarm CSI enthusiasts can gather. I think the linked discord can be that for live chat. Wdyt?

@s4ke I prefer to discuss on GitHub issues so those comments are visible for whole world and does not disappear but you can already find me from [Docker Community Slack](https://dockercommunity.slack.com) if there is some questions which you don't want make public.

> Would it be possible to support [cephfs](https://docs.ceph.com/en/quincy/cephfs/index.html) for self-hosting?

@bluepuma77 I have very little knowledge about CephFS but I would guess that getting CSI driver working is the easy part. Hard part most probably is if you need get Ceph cluster up and running to Swarm.

Personally I prefer to stay away from this kind of overly complex storage systems and use application layer replication instead of (example Redis cluster, RabbitMQ cluster, etc...).

**Comment by bluepuma77** [comment]
> find me from [Docker Community Slack](https://dockercommunity.slack.com)

Seems you can only sign up for "Docker Community Slack" with an "@docker.com" email.

**Comment by olljanat** [comment]
> Seems you can only sign up for "Docker Community Slack" with an "@docker.com" email.

Oh, that is boring. I prefer to stay there because years ago I got invited maintainers private channel (even when I'm not maintainer but was quite active contributor on that time).


Anyway, looks that I'm able to share invite link. This should  be valid next 30 days https://join.slack.com/t/dockercommunity/shared_invite/zt-1p35iok6c-paZhr19gaAlFxP~cqJu_FA

**Comment by neersighted** [comment]
There is a permanent invite at https://dockr.ly/slack.

I would also like to nudge as many people as possible to this issue tracker, and to use it as a central location to coordinate upstream (Moby, SwarmKit) and downstream (plugin) issues.

**Comment by olljanat** [comment]
> There is a permanent invite at https://dockr.ly/slack.

@neersighted it does not work anymore. You will see if you open it on browser in incognito mode.

**Comment by neersighted** [comment]
I'll nudge the relevant people.

**Comment by olljanat** [comment]
> I would also like to nudge as many people as possible to this issue tracker, and to use it as a central location to coordinate upstream (Moby, SwarmKit) and downstream (plugin) issues.

Yes, I was thinking same when started listing know issues and feature requests in here.

**Comment by neersighted** [comment]
http://dockr.ly/comm-slack is evidently the current link, the old one can no longer be updated due to an issue with their URL shortening service.
